java-wasm-allocator/src/main/java/org/m4d/adp/allocator/AllocatorInterface.java

```diff
@@ -14,8 +16,12 @@
 
     static{
         if (!WasmInterface.LOADED_EMBEDDED_LIBRARY) {
-            Path p = Paths.get("../wasm_interface/target/release/libwasm_interface.so");
-            System.load(p.toAbsolutePath().toString());
+            try {
+                System.out.println("nativeLoader");
+                NativeLoader.loadLibrary("wasm_interface");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
         }
```

Comment:
Consider avoiding the duplication, e.g. by putting everything in WasmInterface and removing AllocatorInterface.